site/docs/api/error-handling.md

```diff
@@ -0,0 +1,136 @@
+---
+title: Error Handling Strategies
+description: Ways of coping with unexpected error in Hazelcast Jet.
+---
+
+There is unavoidable risk in running Jet jobs. They ingest data from
+external sources, the correctness and consistency of which can't be
+guaranteed. The processing they perform includes the running of user
+code, which is written by humans and, as such, is unavoidably imperfect.
+
+So even if we assume no bugs in the Jet framework itself, errors can and
+will happen. What options Jet offers for dealing with them is the topic
+of the next sections.
+
+## Scope
+
+A Jet cluster, at any given time, is running an arbitrary number of
+independent jobs in parallel. These jobs do not interact in any way, and
+the failure of one should not lead to any consequences for any other.
+When considering error handling, the basic unit of discussion will be
+the job, mainly what happens with it once it encounters a problem.
+
+The default way Jet reacts to an error is to fail the job that has
+produced it, but what one can do with such a failed job afterward is the
+interesting part.
+
+## Types of Errors
+
+Before we dig deeper, let's list some typical error one might encounter:
+
+* *corrupt data*: ex. user-provided code is parsing JSON lines, but some
+  of them are incorrectly formatted
+* *bugs in user code*: ex. user provided filtering lambda throws a
+  `NullPointerException` for certain input
+* *connection to external services lost*: Jet sink writing into an
+  `IMap` loses connection to the remote Hazelcast cluster hosting the
+  data
```

Comment:
This discrimination is too anecdotal, we can do something more abstract:
```suggestion
* *input data error*: unexpected input data, e.g. unexpected null value, out-of-range date (`0000-00-00`) or malformed JSON document
* *coding errors*: e.g. NullPointerException, memory leaks, ...
* *I/O errors*: network errors, remote service failures
```
Also "ex." -> "for example," or "e.g."